### PR TITLE
Update branding to 3.0.1

### PR DESCRIPTION
**Pull request by dougbu**
- aspnet/AspNetCore-Internal#3153

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

**Comment by dougbu** [comment]
```
F:\workspace\_work\1\s\src\Framework\ref\Microsoft.AspNetCore.App.Ref.csproj(177,5): error MSB3073: The command "tar -czf F:\workspace\_work\1\s\artifacts\installers\Release\aspnetcore-targeting-pack-3.0.1-ci.tar.gz ." exited with code 9009.
```
Either the targeting pack is not written anymore or its path changed. Both would be an unintended side effect of my changes.

**Comment by BrennanConroy** [comment]
That's unrelated to this change. It's happening on a bunch of PRs. @JunTaoLuo and I think it might be an agent issue.

File: eng/targets/ResolveReferences.targets [hunk 1]
```diff
@@ -36,7 +36,8 @@
     -->
     <UseLatestPackageReferences Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsServicingBuild)' != 'true'  ">true</UseLatestPackageReferences>
     <UseLatestPackageReferences Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsImplementationProject)' != 'true' ">true</UseLatestPackageReferences>
-    <UseLatestPackageReferences Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsImplementationProject)' == 'true' AND ( '$(IsServicingBuild)' != 'true' OR '$(IsPackable)' == 'true' ) ">true</UseLatestPackageReferences>
+    <UseLatestPackageReferences
+        Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsImplementationProject)' == 'true' AND '$(IsPackable)' == 'true' ">true</UseLatestPackageReferences>
```
Comment (dougbu) [inline eng/targets/ResolveReferences.targets]: Remaining changes in this file just address nits. Removed of `'$(IsServicingBuild)' != 'true' OR` because that's already checked 2 lines above.

**Comment by dougbu** [comment]
> @JunTaoLuo and I think it might be an agent issue.

Well, exit code 9009 means "file not found". If it's an agent issue, that would mean `tar` isn't found. @MattGal have the agents in NetCorePublic-Pool / BuildPool.Windows.10.Amd64.VS2019.Open moved to an **older** Windows build? (`tar` is now available by default but that's fairly recent.)

@BrennanConroy thanks, this makes more sense than my thought the targeting pack itself wasn't found. The command _creates_ the .tar.gz file from files in its current directory.

**Comment by dougbu** [comment]
This is ready to go once `tar` is found on the agents again

**Comment by dougbu** [comment]
:up::date: to include a more complete dotnet/core-eng#7970 workaround
- may actually build this time 🤞

File: Directory.Build.props [hunk 2]
```diff
@@ -86,7 +86,7 @@
     <TargetingPackInstallerBaseName>aspnetcore-targeting-pack</TargetingPackInstallerBaseName>
 
     <!-- Used to only produce targeting pack installers/packages once per major.minor. -->
-    <IsTargetingPackBuilding Condition="'$(IsServicingBuild)' == 'true' AND '$(DotNetBuildFromSource)' != 'true'">false</IsTargetingPackBuilding>
+    <IsTargetingPackBuilding Condition="'$(AspNetCorePatchVersion)' == '0' AND '$(DotNetBuildFromSource)' != 'true'">false</IsTargetingPackBuilding>
```
Comment (JunTaoLuo) [inline Directory.Build.props]: Hmm well, we need to rebuild the targeting pack in 3.0.1 so this logic needs some finessing. We need to be able to control whether the targeting pack builds. We can use this as a default logic though.

File: build.ps1 [hunk 3]
```diff
@@ -328,10 +330,10 @@ if ($CI) {
 }
 
 # tools.ps1 corrupts global state, so reset these values in case they carried over from a previous build
-rm variable:global:_BuildTool -ea Ignore
-rm variable:global:_DotNetInstallDir -ea Ignore
-rm variable:global:_ToolsetBuildProj -ea Ignore
-rm variable:global:_MSBuildExe -ea Ignore
+Remove-Item variable:global:_BuildTool -ea Ignore
```
Comment (JunTaoLuo) [inline build.ps1]: Why this change? Looks fine but I'm just curious.

**Review by JunTaoLuo — APPROVED** [review]
🥇 for coming up with a workaround to unblock us!

File: build.ps1 (hunk 3 above)
Comment (dougbu) [reply build.ps1]: Because aliases are bad in scripts and VS Code warns about them

**Comment by dougbu** [comment]
Success modulo a missing `New-Item -ItemType Directory .\.tools\ -ErrorAction SilentlyContinue` ❕ ❕
